controllers/authController.js

```diff
@@ -19,57 +19,69 @@ const signUp = async (req , res)  =>{
        res.status(422).json({error:"password must not be less than six characters"})
     }
     
-    await  Users.findOne({email : email})
-    .then(user =>{
-      if(user){
+const Email = await Users.findOne({email : email })
+
+    if (Email){
         res.status(422).json({error:"Email already registered"})
-      } else {
+      } 
 
-const newUser = new Users({name , email , username ,  password , password2})
+const User = await Users.findOne({username : username})
+
+    if(User){
+
+        res.status(422).json({error:"Username already exists!"})
 
-        //Hash password
 
-        bcrypt.genSalt(10 , ( err , salt )=>
+    }  else {
+const newUser = new Users({name , email , username ,  password , password2})
+        //Hash password
+        bcrypt.genSalt(10 , ( err , salt)=>
             bcrypt.hash(newUser.password, salt , async ( err , hash) =>{
-                if (err) throw err;
+                if(err) throw err;
                 newUser.password = hash
                 // save 
                 await newUser.save();
                 res.status(200).json("Registered");
             })
         )
 
+    }
 
-      }
-    })
+    }
+   
     
 
-}
+
 // Sign In
 
-const signIn = async  (req , res ) =>{
+const login = async  (req , res ) =>{
     const {username , password} = req.body;
+
     if(!username  || !password){
+
         res.json(422).json({error:'Incorrect Password!!!'})
     }
     
-    await Users.findOne({username : username})
-    .then(savedUser =>{
-        if(!savedUser){
+const userName =  await Users.findOne({username : username})
+  
+        if(!userName){
             return res.status(422).json({error:"Invalid email or  password"})
         }
-        bcrypt.compare(password, savedUser.password)
-        .then(isMatch =>{
+      
+const isMatch  =  bcrypt.compare(password, userName.password)
+        
             if(isMatch){
-                const token = jwt.sign({_id:savedUser._id}, process.env.JWT_SECRET)
+
+                const token = jwt.sign({_id:userName._id}, process.env.JWT_SECRET)
                 res.status(200).json({token})
```

Comment:
How would the frontend know which user just logged in?

Reply:
The token is the respond

Reply:
The frontend would not decode the token. They would just store it and be sending it back to you. The frontend would need a user object.